File: vko1/vko1/cpp/student.cpp
#include "student.h"
#include <iostream>
#include <string>
using namespace std;

    Student::Student(){
        setName("NONAME");
        setStudentnumber("NONUMBER");
        setCredits(0);
    }

    Student::Student(const string& aName, const string& aStudentnumber){
        setName(aName);
        setStudentnumber(aStudentnumber);
        setCredits(0);
    }

    Student::~Student(){

    }

    void Student::setName(const string& aName){
        name = aName;
    }

    string Student::getName() const{
        return name;
    }

    void Student::setStudentnumber(const string& aStudentnumber){
        studentnumber = aStudentnumber;
    }

    string Student::getStudentnumber() const{
        return studentnumber;
    }

    void Student::setCredits(const int& aCredits){
        credits = aCredits;
    }

    int Student::getCredits() const{
        return credits;
    }

    void Student::printStudentInfo() const{
        std::cout << "Name: " << getName() << "\nStudent number: " << getStudentnumber() << "\nCredits: " << getCredits() << std::endl;
    }

File: vko2/java/java-intellij/src/com/company/Student.java
package com.company;

public class Student {
    private String name;
    private String studentNumber;
    private int credits;

    public Student(){
        System.out.printf("Student constructor%n");
    }
    public Student(String aName, String aStudentNO){
        System.out.printf("Student constructor with arguments%n");
        this.setName(aName);
        this.setStudentNO(aStudentNO);
        this.setCredits(0);
    }

    public void setName(String aName){
        this.name = aName;
    }
    public String getName(){
        return this.name;
    }


    public void setStudentNO(String aStudentNO){
        this.studentNumber = aStudentNO;
    }
    public String getStudentNO(){
        return this.studentNumber;
    }

    public void setCredits(int aCredits){
        this.credits = aCredits;
    }
    public int getCredits(){ return this.credits; }

    public void printInfo(){
        System.out.printf("Name: %s%nStudent number: %s%nCredits: %d%n", name, studentNumber, credits );
    }
}
